src/rns/rns.processor.ts

```diff
@@ -178,15 +192,15 @@ const commands = {
   TokenSold: tokenSoldHandler
 }
 
-function isValidEvent (value: string): value is keyof typeof commands {
+function isValidEvent(value: string): value is keyof typeof commands {
   return value in commands
 }
 
-export default function rnsProcessorFactory (logger: Logger, eth: Eth) {
+export default function rnsProcessorFactory(logger: Logger, eth: Eth, app?: Application) {
   return async function (eventData: EventData): Promise<void> {
     if (isValidEvent(eventData.event)) {
       logger.info(`Processing event ${eventData.event}`)
-      await commands[eventData.event](logger, eventData, eth)
+      await commands[eventData.event](logger, eventData, eth, app)
```

Comment:
Watch out, you have `app` defined as `?` but in the handlers, you use it without the `?` (eq. you don't check if it exists). Fix that, please.

Also, think about how this will work for the `precache` CLI command. To be honest I don't like passing the whole `app` to the processor. Somehow I  would like to keep them separated for the "separation of concerns". The processor should not have to know on what routes are the services defined etc.

What about having an object like this:
```
const services = {domains, offers, sold}
```
and pass that instead of the `app`?

Reply:
sure, that can be done. on the other hand, having the entire context available seems to be the feathers way. That is why I've done it so. But I have no quarrel with passing just the services.
As to the precache, it will not use that, that is why it is optional in the factory. And yes, it does need to check whether it exists and fallback to using the model.

Reply:
Well, one (me :-P) could argue that Cache consists of two parts blockchain layer listening **and processing** events from blockchain and then presentation layer that is built with FeathersJs. These two layers are "connected" through DB (with introducing this change this becomes less true, unfortunately), but generally these two layers should be kept separate.

Also, you are wrong that `precache` won't use it. It will ;-) It uses the same processor and also needs a way how to store the the data...